### PR TITLE
Checks extension

### PR DESCRIPTION
**Pull request by alexander-yevsyukov**
This PR continues improvements to the code generation infrastructure, addressing mostly naming. 

In particular:
  * The package `io.spine.tools.protodoc` was transformed to `io.spine.tools.javadoc.style`.
     The Gradle plugin class and other classes of the package was renamed accordingly.
  * Gradle project extension classes and task name enums were renamed after short plugin names (e.g. `McDartExtension`, `McDartTaskName`).
  * `ValidatingBuilderWhich` was renamed to `MessageBuilderWhich` because the class deals with `Message.Builder`s.
  * `GeneratedValidatingBuilderWhich` was renamed to `ValidatingBuilderWhich` to save some pixels.
  * `ProjectArguments` class was de-utilised and renamed to `JavaCompileTasks`. It's the only thing it did and does.
  * The abstract base for JS-related code-generation tasks was renamed from `GenerationTask` to `CodeGenStep` to avoid the name clash with Gradle `Task`s. Our code of the steps does not deal with Gradle.

 Also, the McDart plugin now supports resolving imports in generated tests sources too.

### CODE REVIEW

File: tools/mc-java/src/main/java/io/spine/tools/mc/java/gradle/JavaCompileTasks.java [hunk 1]
```diff
@@ -38,29 +38,30 @@
...
+    private JavaCompileTasks(Project project) {
+        TaskContainer allTasks = project.getTasks();
+        this.tasks = allTasks.withType(JavaCompile.class);
+    }
+
+    /**
+     * Creates a new instance for the passed projects.
+     */
+    static JavaCompileTasks of(Project project) {
+        return new JavaCompileTasks(project);
```
Comment (armiol) [inline tools/mc-java/src/main/java/io/spine/tools/mc/java/gradle/JavaCompileTasks.java:56]: I would check the `project` for not being `null`.

File: tools/javadoc-style/src/test/java/io/spine/tools/javadoc/style/BacktickFormattingTest.java [hunk 2]
```diff
@@ -70,11 +70,13 @@ void not_handle_multi_lined_text_surrounded_with_backticks() {
     }
 
     /**
+     * Tests that formatting correctly handles lines containing the dollar sign.
```
Comment (dmdashenkov) [inline tools/javadoc-style/src/test/java/io/spine/tools/javadoc/style/BacktickFormattingTest.java:73]:
```suggestion
     * Tests that the formatter handles lines containing the dollar sign correctly.
```

File: tools/mc-java/src/test/java/io/spine/tools/gradle/SpinePluginTest.java [hunk 3]
```diff
@@ -57,8 +58,16 @@
...
 
-@DisplayName("SpinePluginBuilder should")
-class SpinePluginBuilderTest {
+/**
+ * Tests {@link SpinePlugin}.
+ *
+ * @apiNote This test suite is placed under the {@code mc-java} module, while the class
+ * it tests ({@link SpinePlugin} is under {@code }) because the test suite uses
+ * {@link McJavaTaskName}. Presumably, this test suite should be updated using
+ * stub task names (which would be language-neutral) and returned back to the
```
Comment (armiol) [inline tools/mc-java/src/test/java/io/spine/tools/gradle/SpinePluginTest.java:67]: "... future."?

**Review by armiol — APPROVED** [review]
@alexander-yevsyukov LGTM with a few comments to address.

File: tools/mc-js/src/main/java/io/spine/tools/mc/js/code/step/ResolveImports.java [hunk 4]
```diff
@@ -44,10 +44,10 @@
  *
  * <p>Supports only {@code CommonJS} imports.
  *
- * <p>The task should be performed last among {@linkplain GenerationTask generation tasks}
+ * <p>The task should be performed last among {@linkplain CodeGenStep generation tasks}
```
Comment (dmdashenkov) [inline tools/mc-js/src/main/java/io/spine/tools/mc/js/code/step/ResolveImports.java:47]:
```suggestion
 * <p>This step should be performed last among {@linkplain CodeGenStep code generation steps}
```

File: tools/mc-js/src/main/java/io/spine/tools/mc/js/code/step/ResolveImports.java [hunk 5]
```diff
@@ -44,10 +44,10 @@
...
  * to ensure that imports won't be modified after execution of this task.
```
Comment (dmdashenkov) [inline tools/mc-js/src/main/java/io/spine/tools/mc/js/code/step/ResolveImports.java:48]:
```suggestion
 * to ensure that imports won't be modified later.
```

**Review by dmdashenkov — APPROVED** [review]
@alexander-yevsyukov, LGTM with a couple of doc-related comments.

**Pull request merged**